### PR TITLE
[fix](hive)Modify the Hive notification event processing method when using meta cache and add parameters to the Hive catalog.

### PR DESCRIPTION
**Pull request by hubgeter**
before #15401  and #38244 

## Proposed changes

1. Add the parameter `hive.enable_hms_events_incremental_sync` to the hive catalog, which is used to switch the catalog to read hive notification events (default is false). The default value is `enable_hms_events_incremental_sync` in fe.conf 

2. Add the parameter `hive.hms_events_batch_size_per_rpc` to the hive catalog, which is used to set the size of notification events read by the catalog each time. The default value is `hms_events_batch_size_per_rpc` in fe.conf (default is 500)

3. append hms event notification case . 

4. Remove the `use_meta_cache` setting in catalog that is forced to true.

Example :
```
create catalog if not exists catalog_name properties (
      "type"="hms",
      'hive.metastore.uris' = 'thrift://externalEnvIp:hms_port',
       "hive.enable_hms_events_incremental_sync" ="true",
       "hive.hms_events_batch_size_per_rpc" = "1000"
);
```

### CODE REVIEW

**Comment by hubgeter** [comment]
run buildall

**Comment by hubgeter** [comment]
run buildall

**Comment by hubgeter** [comment]
run buildall

**Comment by hubgeter** [comment]
run buildall

**Comment by hubgeter** [comment]
run buildall

**Comment by hubgeter** [comment]
run buildall

**Comment by hubgeter** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/hive/event/MetastoreEventsProcessor.java [hunk 1]
```diff
@@ -125,6 +128,8 @@ private void realRun() {
                 } catch (MetastoreNotificationFetchException e) {
                     LOG.warn("Failed to fetch hms events on {}. msg: ", hmsExternalCatalog.getName(), e);
                 } catch (Exception ex) {
+                    hmsExternalCatalog.onRefresh(true);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/hive/event/MetastoreEventsProcessor.java:131]: What is this 2 lines for?

File: fe/fe-core/src/main/java/org/apache/doris/datasource/hive/event/AlterPartitionEvent.java [hunk 2]
```diff
@@ -145,9 +145,7 @@ protected boolean canBeBatched(MetastoreEvent that) {
         // `that` event can be batched if this event's partitions contains all of the partitions which `that` event has
         // else just remove `that` event's relevant partitions
         for (String partitionName : getAllPartitionNames()) {
-            if (thatPartitionEvent instanceof AddPartitionEvent) {
-                ((AddPartitionEvent) thatPartitionEvent).removePartition(partitionName);
-            } else if (thatPartitionEvent instanceof DropPartitionEvent) {
+            if (thatPartitionEvent instanceof DropPartitionEvent) {
```
[the remark this answers is not in the document]
Comment (hubgeter) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/hive/event/AlterPartitionEvent.java:148]: Suppose you have a Hive partition table `users`. Execute the `INSERT INTO TABLE users PARTITION (country='ABC') VALUES (3, 'Rahul Kumar', 28), (4, 'Priya Singh', 24);` statement. If you did not have the partition `country='ABC'` before, Hive will generate two events.

1. add partition event
2. alter partition event

The original logic is to merge these two events into one alter event for processing, but in the end this partition will not be added to the partitions cache. In order to reduce the judgment logic of the alter event, I will not merge these two events here.

File: fe/fe-core/src/main/java/org/apache/doris/datasource/hive/event/MetastoreEventsProcessor.java (hunk 1 above)
Comment (hubgeter) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/hive/event/MetastoreEventsProcessor.java:131]: Suppose you receive a set of events, and when processing the insert event, the initialization reports a nullpointer. The event processing fails, and our data with Hive is out of sync? Refresh it and you can process other events next time. i will modify it to `onRefreshCache`

File: fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HMSExternalCatalog.java [hunk 3]
```diff
@@ -101,6 +106,12 @@ public void checkProperties() throws DdlException {
                     "The parameter " + FILE_META_CACHE_TTL_SECOND + " is wrong, value is " + fileMetaCacheTtlSecond);
         }
 
+        enableHmsEventsIncrementalSync =
+                catalogProperty.getOrDefault(HMSProperties.ENABLE_HMS_EVENTS_INCREMENTAL_SYNC, "false").equals("true");
```
[the remark this answers is not in the document]
Comment (hubgeter) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HMSExternalCatalog.java:110]: I will modify it

File: fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HMSExternalCatalog.java [hunk 4]
```diff
@@ -101,6 +106,12 @@ public void checkProperties() throws DdlException {
...
+
+        hmsEventsBatchSizePerRpc =
+                Integer.valueOf(catalogProperty.getOrDefault(HMSProperties.HMS_EVENTIS_BATCH_SIZE_PER_RPC, "-1"));
```
[the remark this answers is not in the document]
Comment (hubgeter) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HMSExternalCatalog.java:113]: Originally I wanted to indicate that this value was not set, but now I want its default value to be `Config.hms_events_batch_size_per_rpc`

**Comment by hubgeter** [comment]
run buildall

**Comment by hubgeter** [comment]
run buildall

**Comment by hubgeter** [comment]
run p1

**Review by morningman — DISMISSED** [review]
LGTM

**Review by zddr — APPROVED** [review]

**Review by CalvinKirs — DISMISSED** [review]

**Comment by hubgeter** [comment]
run buildall

**Comment by hubgeter** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Comment by hubgeter** [comment]
run buildall

**Review by wuwenchi — APPROVED** [review]

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**